src/transport/SecureSessionMgr.h

```diff
@@ -182,17 +184,23 @@ class DLL_EXPORT SecureSessionMgr : public TransportMgrDelegate
 
     /**
      * @brief
-     *   Send a message to a currently connected peer.
+     *   This function takes the payload and returns the final message which can be send multiple times.
      *
      * @details
-     *   msgBuf contains the data to be transmitted.  If bufferRetainSlot is not null and this function
-     *   returns success, the encrypted data that was sent, as well as various other information needed
-     *   to retransmit it, will be stored in *bufferRetainSlot.
+     *   It contains following preparation:
+     *    1. Encrypt the msgBuf
+     *    2. construct the packet header
+     *    3. Encode the packet header and prepend it to message.
+     *   Returns a prepared message in preparedMessage.
+     */
+    CHIP_ERROR PrepareMessage(SecureSessionHandle session, PayloadHeader & payloadHeader, System::PacketBufferHandle && msgBuf,
```

Comment:
The way I understand it, the method takes a buffer of data and encrypts its + adds the packet header. I find "prepare" a bit vague and it forces me to read through the comments to understand what the method does.

Is it correct that this is the one way we expect to convert a packet handle to an encrypted one? How about  in this case:

`EncryptPayload` or `BuildEncryptedMessagePayload`

and 

`SendEncryptedPayload` ?

This way I do not read comments, I understand that giving a buffer, I encrypt it and send it.

Reply:
This API is also handling unsecure message in SessionEstablishmentExchangeDispatch.cpp, So it is not proper to name it `BuildEncryptedMessagePayload`

Reply:
The prepared steps in the comment say "it encrypts the msgBuf". 
I am trying to find out if we can get a more descriptive name than "Prepare". From what I see this is that this has:
  - Input: session, header, messagebuffer
  - Output: EncryptedMessageBuffer
  
  Which tells me it encrypts and adds a header. Realistically also wondering why we take a sessionhandle instead of just keys if we do encryption, but that is not for this PR.
  
  Could you either update the comment about encryption or find some descriptive name (or both)?

Reply:
I think missing "override" made me not realize this particular implementation does "encrypt" but others may not. Please add override keyword here and below as needed and then I think the name is ok.

Reply:
Actually, this particular impl always encrypts.  The message dispatch bits can keep Prepare(), etc, but we could name this API something different and clearer.